src/ui/business/products/products.ui-business.ts

```diff
@@ -0,0 +1,17 @@
+import { Page } from "@playwright/test";
+import { AddNewProductPage } from "ui/pages/products/addNewProduct.page";
+import { ProductsPage } from "ui/pages/products/product.page";
+
+export class ProductsUIService {
+    private productsPage: ProductsPage;
+    private addNewProductPage: AddNewProductPage;
+
+    constructor(private page: Page) {
+        this.productsPage = new ProductsPage(page);
+        this.addNewProductPage = new AddNewProductPage(page);
+    }
```

Comment:
Для каждого сервиса повторяется создание конструктора
Как вариант можно создать `class PageHolder` от которого будут наследоваться классы, которым нужна `Page` 

```suggestion
class PageHolder {
      constructor(protected page: Page) {}
}

export class ProductsUIService {
    private productsPage: ProductsPage = new ProductsPage(this.page);
    private addNewProductPage: AddNewProductPage  = new AddNewProductPage(this.page);
    }
```